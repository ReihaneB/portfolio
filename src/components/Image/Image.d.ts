import { ImageProps as NextImageProps, StaticImageData } from 'next/image';

export interface ImageProps extends NextImageProps {
  width?: number;
  height?: number;
  src: StaticImageData | string;
  alt: string;
}
