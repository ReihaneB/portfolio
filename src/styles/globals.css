:root {
  /* colors */
  --white: #f5f5f7;
  --cyan: #06b6d4;
  --blue: #2756df;
  --blue-lighter: #cee5ff;
  --blue-light: #dbeafe;
  --blue-dark: #1e40af;
  --focus-blue: #93c5fd;
  --grey: #6b7280;
  --grey-light: #e5e7eb;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  border: 0;
  font: inherit;
  vertical-align: baseline;
}

html,
body {
  max-width: 100vw;
  overflow-x: hidden;
}

details,
figcaption,
figure,
footer,
hgroup,
menu,
nav {
  display: block,
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote::before,
blockquote::after,
q::before,
q::after {
  content: '';
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

a {
  color: inherit;
  text-decoration: none;
}

h1 {
  font-size: 30px;
  line-height: 38px;
  font-weight: bold;
}

h2 {
  font-size: 20px;
  line-height: 24px;
  font-weight: bold;
}

h3 {
  font-size: 16px;
  line-height: 20px;
  font-weight: bold;
}

h4 {
  font-size: 14px;
  line-height: 18px;
}

h5 {
  font-size: 12px;
  line-height: 16px;
}

p {
  font-size: 16px;
  line-height: 24px;
}