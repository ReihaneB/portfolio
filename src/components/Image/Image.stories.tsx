import type { Meta, StoryObj } from '@storybook/react';

import Square from '@/styles/assets/placeholder/1x1.png';
import _4x3 from '@/styles/assets/placeholder/4x3.png';
import _16x9 from '@/styles/assets/placeholder/16x9.png';

import Image from './Image';

import styles from './Image.stories.module.css';

const meta: Meta<typeof Image> = {
  title: 'Components/Image',
  component: Image,
  decorators: [
    Story => (
      <div className={styles.root}>
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Image>;

const defaultArgs = {
  src: _16x9,
  alt: 'Placeholder image',
  priority: false,
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const SquareImage: Story = {
  args: {
    ...defaultArgs,
    src: Square,
  },
};

export const _4x3Image: Story = {
  args: {
    ...defaultArgs,
    src: _4x3,
  },
};

export const _16x9Image: Story = {
  args: {
    ...defaultArgs,
    src: _16x9,
  },
};
