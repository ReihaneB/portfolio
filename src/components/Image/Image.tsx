import NextImage from 'next/image';

import styles from './Image.module.css';
import type { ImageProps } from './Image.d';

function Image({
  id,
  src,
  alt,
  ...rest
}: ImageProps) {
  return (
    <NextImage
      id={id}
      src={src}
      alt={alt}
      className={styles.image}
      width={0}
      height={0}
      {...rest}
    />
  );
}

export default Image;
