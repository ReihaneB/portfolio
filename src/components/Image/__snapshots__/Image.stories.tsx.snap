// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Image _4x3Image smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <img alt="Placeholder image"
       loading="lazy"
       width="800"
       height="600"
       decoding="async"
       data-nimg="1"
       class="src-components-Image-Image-module__image--gXjYq"
       srcset="/static/media/_4x3.58bee57f.png?w=828&amp;q=75 1x, /static/media/_4x3.58bee57f.png?w=1920&amp;q=75 2x"
       src="/static/media/_4x3.58bee57f.png?w=1920&amp;q=75"
       style="color: transparent;"
  >
</main>
`;

exports[`Components/Image _16x9Image smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <img alt="Placeholder image"
       loading="lazy"
       width="1920"
       height="1080"
       decoding="async"
       data-nimg="1"
       class="src-components-Image-Image-module__image--gXjYq"
       srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
       src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
       style="color: transparent;"
  >
</main>
`;

exports[`Components/Image Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <img alt="Placeholder image"
       loading="lazy"
       width="1920"
       height="1080"
       decoding="async"
       data-nimg="1"
       class="src-components-Image-Image-module__image--gXjYq"
       srcset="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=3840&amp;q=75 2x"
       src="/static/media/_16x9.d50e36d3.png?w=3840&amp;q=75"
       style="color: transparent;"
  >
</main>
`;

exports[`Components/Image SquareImage smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <img alt="Placeholder image"
       loading="lazy"
       width="1080"
       height="1080"
       decoding="async"
       data-nimg="1"
       class="src-components-Image-Image-module__image--gXjYq"
       srcset="/static/media/_1x1.ea48a62d.png?w=1080&amp;q=75 1x, /static/media/_1x1.ea48a62d.png?w=3840&amp;q=75 2x"
       src="/static/media/_1x1.ea48a62d.png?w=3840&amp;q=75"
       style="color: transparent;"
  >
</main>
`;
